driver-sync/src/main/com/mongodb/client/internal/KeyManagementService.java

```diff
@@ -98,8 +101,21 @@ private void enableHostNameVerification(final SSLSocket socket) {
     private void closeSocket(final Socket socket) {
         try {
             socket.close();
-        } catch (IOException e) {
+        } catch (IOException | RuntimeException e) {
             // ignore
         }
     }
+
+    /**
+     * @return Never.
+     */
+    private static RuntimeException translateInterruptedExceptionAndThrow(final IOException e, final String message) throws IOException,
+            MongoInterruptedException {
+        Optional<MongoInterruptedException> interruptedException = translateInterruptedException(e, message);
+        if (interruptedException.isPresent()) {
```

Comment:
Why can't we use `orElseGet` pattern here as in Crypt?

Reply:
We can if we change `KeyManagementService.stream` from `throws IOException` to `throws Exception`. I don't want to do that. Fortunately, if we do https://github.com/mongodb/mongo-java-driver/pull/1204/files#r1337520600, this change in `KeyManagementService` will be gone.